functions/Get-DbaTable.ps1

```diff
@@ -76,22 +76,43 @@ Returns information on the CommandLog table in the DBA database on both instance
 		
 		if ($Table) {
 			foreach ($t in $Table) {
-				$dotcount = ([regex]::Matches($t, "\.")).count
-				
+                $splitName = [regex]::Matches($t, "(\[.+?\])|([^\.]+)").Value
```

Comment:
OK,
now I need two lines to replace those 35:

> $test = $case -match '\[?(?<one>[^\]]*)?\]?.?\[?(?<two>[^\]]*)?\]?.?\[?(?<three>[^\]]*)\]?'
$database,$schema,$Table = switch (!$Matches.three + !$Matches.two + !$Matches.one){ 2 { $null, $null, $matches.one ;break } 1 { $null, $matches.one, $matches.two ;break } default { $matches.one, $matches.two, $matches.three }}

😄 
well, that handles the regex, the dotcount, the variable assignments and the [] replacements,
a catch for invalid parameter may still be needed